e2e/workloads/workloads.go

```diff
@@ -5,19 +5,54 @@ package workloads
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/ramendr/ramen/e2e/config"
 	"github.com/ramendr/ramen/e2e/types"
 )
 
-type factory func(revision string, pvcSpec config.PVCSpec) types.Workload
+// factoryFunc is the new() function type for workloads
+type factoryFunc func(revision string, pvcSpec config.PVCSpec) types.Workload
 
-var registry = map[string]factory{
-	deploymentName: NewDeployment,
+// Registry maintains a map of workload types to their factory functions.
+type Registry struct {
+	mutex     sync.RWMutex
+	workloads map[string]factoryFunc
+}
```

Comment:
Using a type is not needed, if we need to make this thread safe, we can add a mutex in the package, protecting the map. Why do we need to add a type for this? It does not provide any protection since any code in the package can access registery.deployers directly.